plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/OTelResourceProvider.java

```diff
@@ -92,6 +99,8 @@ private static SdkMeterProvider createSdkMetricProvider(Settings settings, Resou
                     .setInterval(TelemetrySettings.METRICS_PUBLISH_INTERVAL_SETTING.get(settings).getSeconds(), TimeUnit.SECONDS)
                     .build()
             )
+            .registerView(InstrumentSelector.builder().setName("*"+ DYNAMIC_HISTOGRAM_METRIC_NAME_SUFFIX).setType(InstrumentType.HISTOGRAM).build(), View.builder()
```

Comment:
Why do we need that?

Reply:
Trying to understand here, we want the histograms without explicit buckets should be exponential one. Why do we need to play with histogram namings, could we use InstrumentSelector to apply by default to `*` (for histograms) unless it has explicit buckets? (I am wondering if that shouldn't be the default behaviour)

Reply:
Unfortunately, InstrumentSelector works on only a few things name, unit and type. We can not identify if the instrument has explicit buckets defined through the InstrumentSelector. Also, ExplicitBuckets are defined as an advice to the histogram.

Reply:
I also didn't like implementing this way. Let me see if anything else can be done here.

Reply:
There is one other way where we can internally ask the different [OtelMeter](https://opentelemetry.io/docs/specs/otel/metrics/api/#meter) from [MeterProvider](https://opentelemetry.io/docs/specs/otel/metrics/api/#meterprovider) for exponential histogram and configure the view with separate aggregation for this meter. Meter means it's going to create a separate [instrumentation scope](https://opentelemetry.io/docs/specs/otel/glossary/#instrumentation-scope) something like this 

```
{
    "resourceMetrics": [{
        "resource": {
            "attributes": [{
                "key": "service.name",
                "value": {
                    "stringValue": "OpenSearch"
                }
            }]
        },
        "scopeMetrics": [{
            "scope": {
                "name": "org.opensearch.telemetry"
            },
            "metrics": [{
                "name": "TestLatencyFixed",
                "unit": "ms",
                "histogram": {
                    "dataPoints": [{
                        "startTimeUnixNano": "1707058608181736000",
                        "timeUnixNano": "1707058668189025000",
                        "count": "5",
                        "sum": 2760,
                        "bucketCounts": ["0", "1", "0", "0", "4"],
                        "explicitBounds": [1, 10, 30, 100],
                        "min": 5,
                        "max": 947
                    }],
                    "aggregationTemporality": 2
                }
            }]
        }, {
            "scope": {
                "name": "org.opensearch.telemetry.dynamic"
            },
            "metrics": [{
                "name": "TestLatency",
                "unit": "ms",
                "exponentialHistogram": {
                    "dataPoints": [{
                        "startTimeUnixNano": "1707058608181736000",
                        "timeUnixNano": "1707058668189025000",
                        "count": "5",
                        "sum": 2760,
                        "scale": 4,
                        "positive": {
                            "offset": 37,
                            "bucketCounts": ["1", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "1", "0", "0", "0", "0", "0", "1", "0", "0", "0", "0", "0", "0", "0", "1", "0", "0", "0", "1"]
                        },
                        "negative": {},
                        "min": 5,
                        "max": 947
                    }],
                    "aggregationTemporality": 2
                }
            }]
        }]
    }]
}
```

Here you can see we have two scopes `org.opensearch.telemetry`  and `org.opensearch.telemetry.dynamic`. `org.opensearch.telemetry.dynamic` (we can name it better) contains the exponential histogram metrics. From code perspective it looks clean but overall if you see the definition of InstrumentationScope it doesn't go well.

**InstrumentationScope**

A logical unit of the application code with which the emitted telemetry can be associated. It is typically the developer’s choice to decide what denotes a reasonable instrumentation scope. The most common approach is to use the [instrumentation library](https://opentelemetry.io/docs/specs/otel/glossary/#instrumentation-library) as the scope, however other scopes are also common, e.g. a module, a package, or a class can be chosen as the instrumentation scope.

Reply:
Thanks @Gaganjuneja , to me any approach with Otel (to cover explicit and implicit bucketing in  way to not surprise consumers) is not looking good. I would suggest at this moment only offer the API with explicit bucketing - at least with that the consumers will get exactly what they want to.

Reply:
@reta Thanks, I agree this is complicating the stuff. While choosing one I think we should go for the `explicit histogram` approach as it automatically decides the bucketing,  low on memory and free the developers from the ambiguity (it's very hard to choose the right explicit bucketing during the develpment phase) of choosing the buckets explicitly. If the request arises for explicit bucketing we can think of providing that feature.

Reply:
:+1: thanks @Gaganjuneja 

> If the request arises for explicit bucketing we can think of providing that feature

You mean `implicit`, right?

Reply:
@reta I am of the option to go with implicit first and if there is requirement where developer wants to take control over and define the granular explicit buckets then we will think.


Otel's implicit exponential bickering logic scales well and automatically tries to readjust the bickering based on the data input and these buckets we can easily be merged.